### PR TITLE
add dpk_connector to dpk

### PR DESCRIPTION
**Pull request by hmtbr**
## Why are these changes needed?

We want to cover the full life cycle for data acquisition and data processing in DPK. This PR will add the core library to perform data acquisition from websites.

## Related issue number (if any).

#633

### CODE REVIEW

File: extensions/bluecrawl_connector/README.md
Comment (daw3rd) [inline extensions/bluecrawl_connector/README.md:1]: Can you add how to use this once it is built.  
If this is is a command lline utility perhaps it belongs in tools.

**Review by daw3rd — DISMISSED** [review]
Can this be scaled up by implementing it as a transform.

**Comment by touma-I** [comment]
> Can this be scaled up by implementing it as a transform.

No. The plan is to release this as an open source package from which we can create additional transform. @daw3rd please remove your request for change. Thanks

File: dpk_connector/Makefile [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Use make help, to see the available rules
+REPOROOT=..
+include $(REPOROOT)/.make.defaults
+
+TAG := "v${DPK_LIB_VERSION}"
```
Comment (daw3rd) [inline dpk_connector/Makefile]: What is this (i.e. TAG) used for?  I don't see any use anywhere.

File: extensions/Makefile
Comment (daw3rd) [inline extensions/Makefile:1]: I think this Makefile and extensions dir can be removed

**Review by daw3rd — CHANGES_REQUESTED** [review]
It would be nice to have some documentation and/or examples to show how this is used.

File: .github/workflows/test-dpk-connector.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+on:
+    workflow_dispatch:
+    push:
+        branches:
+            - "dev"
+            - "releases/**"
+        tags:
+            - "*"
+        paths:
+            - "dpk_connector/**"
```
Comment (daw3rd) [inline .github/workflows/test-dpk-connector.yml]: Probable need this (in general) 

```
            - ".make.*"
```

**Review by daw3rd — CHANGES_REQUESTED** [review]

File: dpk_connector/Makefile (hunk 1 above)
Comment (hmtbr) [reply dpk_connector/Makefile]: Removed.

File: extensions/Makefile
Comment (hmtbr) [reply extensions/Makefile:1]: Removed.

File: .github/workflows/test-dpk-connector.yml (hunk 2 above)
Comment (hmtbr) [reply .github/workflows/test-dpk-connector.yml]: Added.

**Comment by hmtbr** [comment]
@daw3rd Thank you for reviewing this. I added `overview.md` to describe example usage. Could you take a look at the documentation? Let me know if I need additional updates.

**Review by yuanchi2807 — APPROVED** [review]
Thank you for your contribution to DPK.

File: dpk_connector/README.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# DPK Connector
+
+DPK Connector is a scalable and compliant web crawler developed for data acquisition towards LLM development. It is built on [Scrapy](https://scrapy.org/).
+
```
Comment (daw3rd) [inline dpk_connector/README.md]: Can you link to the overview.md?  As is, I can't tell if this is a library, a main() or how to use it.

**Review by daw3rd — CHANGES_REQUESTED** [review]
In order to make the repo feel a bit more cohesive, should we rename this top-level directory to `data-connector-lib` to align with `data-processing-lib`?

File: dpk_connector/README.md (hunk 3 above)
Comment (hmtbr) [reply dpk_connector/README.md]: Added the link. `How to use` also has a link to it.

**Comment by hmtbr** [comment]
> In order to make the repo feel a bit more cohesive, should we rename this top-level directory to data-connector-lib to align with data-processing-lib?

Renamed the top level folder to `data-connector-lib`.

File: .github/workflows/test-dpk-connector.yml
Comment (daw3rd) [inline .github/workflows/test-dpk-connector.yml:1]: Can we rename this to test-connector-lib.yml and the existing test-lib.yml to test-processor-lib.yml?

**Review by daw3rd — CHANGES_REQUESTED** [review]
Thanks for all the changes.  Just 1 more :).

File: .github/workflows/test-dpk-connector.yml
Comment (daw3rd) [reply .github/workflows/test-dpk-connector.yml:1]: I took care of this one.

**Review by daw3rd — APPROVED** [review]
Thanks for the changes!

**Comment by hmtbr** [comment]
Thank you @daw3rd !

**Pull request merged**